src/loaders/OMETiffLoader.js

```diff
@@ -0,0 +1,322 @@
+   * @param {Array} loaderSelection, Array of number Arrays specifying channel selections
+   * @returns {Object} data: TypedArray[], width: number, height: number
+   */
+  async getRaster({ z, loaderSelection }) {
+    const {
+      tiff,
+      offsets,
+      omexml,
+      isLegacyBioFormatsPyramid,
+      poolOrDecoder
+    } = this;
+    const { SizeZ, SizeT, SizeC } = omexml;
+    const rasters = await Promise.all(
+      loaderSelection.map(async index => {
+        const pyramidIndex = z * SizeZ * SizeT * SizeC + index;
+        if (isLegacyBioFormatsPyramid) {
+          if (offsets) {
+            tiff.ifdRequests[pyramidIndex] = tiff.parseFileDirectoryAt(
+              offsets[pyramidIndex]
+            );
+          }
+        } else {
+          const parentImage = await tiff.getImage(index);
+          if (z !== 0) {
+            tiff.ifdRequests[pyramidIndex] = tiff.parseFileDirectoryAt(
+              parentImage.fileDirectory.SubIFDs[z - 1]
+            );
+          }
+        }
+        const image = await tiff.getImage(pyramidIndex);
+        // Flips bits for us for endianness.
+        const raster = await image.readRasters({ pool: poolOrDecoder });
+        return raster[0];
+      })
+    );
+    // Get first selectioz * SizeZ * SizeT * SizeC + loaderSelection[0]n size as proxy for image size.
+    const image = await tiff.getImage(
+      z * SizeZ * SizeT * SizeC + loaderSelection[0]
+    );
+    const width = image.getWidth();
+    const height = image.getHeight();
+    return { data: rasters, width, height };
+  }
+
+  /**
+   * Returns image width and height (at pyramid level z) without fetching data.
+   * This information is inferrable from the provided omexml.
+   * This is only used by the OverviewLayer for inferring the box size.
+   * It is NOT the actual pixel-size but rather the image size
+   * without any padding.
+   * @param {number} z positive integer (0 === highest zoom level)
+   * @returns {Object} width: number, height: number
+   */
+  getRasterSize({ z }) {
+    const { width, height } = this;
+    /* eslint-disable no-bitwise */
+    return {
+      height: height >> z,
+      width: width >> z
+    };
+    /* eslint-disable no-bitwise */
+  }
+
+  async _getChannel({ image, x, y }) {
+    const tile = await image.getTileOrStrip(x, y, 0, this.poolOrDecoder);
+    const is8Bits = image.fileDirectory.BitsPerSample[0] === 8;
+    const is16Bits = image.fileDirectory.BitsPerSample[0] === 16;
+    const is32Bits = image.fileDirectory.BitsPerSample[0] === 32;
+    const data =
+      (is8Bits && new Uint8Array(tile.data)) ||
+      (is16Bits && new Uint16Array(tile.data)) ||
+      (is32Bits && new Uint32Array(tile.data));
+    // Javascript needs little endian byteorder, so we flip if the data is not.
+    // eslint-disable-next-line no-unused-expressions
+    if (!image.littleEndian) {
+      flipEndianness(data);
+    }
+    return data;
+  }
```

Comment:
Isn't `dtype` already a property on `OMETiffLoader`? why do we need to check bitspersample at all?

Reply:
Good point.